Content.Server/Nyanotrasen/Shipyard/Systems/ShipyardSystem.cs

```diff
@@ -0,0 +1,202 @@
+            {
+                _sawmill.Error($"Unable to spawn shuttle {shuttlePath}");
+                return null;
+            };
+
+            _shuttleIndex += _mapManager.GetGrid(gridList[0]).LocalAABB.Width + ShuttleSpawnBuffer;
+            var actualGrids = new List<EntityUid>();
+            var gridQuery = GetEntityQuery<MapGridComponent>();
+
+            foreach (var ent in gridList)
+            {
+                if (!gridQuery.HasComponent(ent))
+                    continue;
+
+                actualGrids.Add(ent);
+            };
+
+            //only dealing with 1 grid at a time for now, until more is known about multi-grid drifting
+            if (actualGrids.Count != 1)
+            {
+                _sawmill.Error($"Unable to spawn shuttle {shuttlePath}");
+                return null;
+            };
+
+            return actualGrids[0];
+        }
+
+        /// <summary>
+        /// Checks a shuttle to make sure that it is docked to the given station, and that there are no lifeforms aboard. Then it appraises the grid, outputs to the server log, and deletes the grid
+        /// </summary>
+        /// <param name="stationUid">The ID of the station that the shuttle is docked to</param>
+        /// <param name="shuttleUid">The grid ID of the shuttle to be appraised and sold</param>
+        public void SellShuttle(EntityUid stationUid, EntityUid shuttleUid, out int bill)
+        {
+            bill = 0;
+
+            if (!TryComp<StationDataComponent>(stationUid, out var stationGrid) || !HasComp<ShuttleComponent>(shuttleUid) || !TryComp<TransformComponent>(shuttleUid, out var xform) || ShipyardMap == null)
+                return;
+
+            var targetGrid = _station.GetLargestGrid(stationGrid);
+
+            if (targetGrid == null)
+                return;
+
+            var gridDocks = _shuttle.GetDocks((EntityUid) targetGrid);
+            var shuttleDocks = _shuttle.GetDocks(shuttleUid);
+            var isDocked = false;
+
+            foreach (var shuttleDock in shuttleDocks)
+            {
+                foreach (var gridDock in gridDocks)
+                {
+                    if (shuttleDock.DockedWith == gridDock.Owner)
+                    {
+                        isDocked = true;
+                        break;
+                    };
+                };
+                if (isDocked)
+                    break;
+            };
+
+            if (!isDocked)
+            {
+                _sawmill.Warning($"shuttle is not docked to that station");
+                return;
+            };
+
+            var mobQuery = GetEntityQuery<MobStateComponent>();
+            var xformQuery = GetEntityQuery<TransformComponent>();
+
+            if (_cargo.FoundOrganics(shuttleUid, mobQuery, xformQuery))
+            {
+                _sawmill.Warning($"organics on board");
+                return;
+            };
```

Comment:
Players can't activate this, only admins can

Reply:
Also iirc the organics function _does_ give feedback in itself but the isdocked doesn't, I will have to implement that here and downstream

Reply:
Ok so I had all of the player feedback in ShipyardSystem.Console and it was in TrySellVessel command, reading info from the event caller (on the button press)

I ripped most of that out since none of it really applies in this implementation. I could implement some theoretical feedback for it if you'd like, otherwise I'm just logging the fail states to console for admin feedback